pkg/sql/schemachanger/scplan/internal/rules/current/dep_drop_object.go

```diff
@@ -115,13 +128,27 @@ func init() {
 		func(from, to NodeVars) rel.Clauses {
 			return rel.Clauses{
 				from.TypeFilter(rulesVersionKey, isDescriptor),
-				to.TypeFilter(rulesVersionKey, isSimpleDependent),
+				to.TypeFilter(rulesVersionKey, isSimpleDependent, Not(isDescriptorBackReference)),
 				JoinReferencedDescID(to, from, "desc-id"),
 				StatusesToAbsent(from, scpb.Status_DROPPED, to, scpb.Status_ABSENT),
 			}
 		},
 	)
 
+	registerDepRule(
```

Comment:
Comment that we don't like these elements and this is to deal with that fact by tightly coupling them to the descriptor?

Can you take a stab at tightly coupling these to the name rather than to the descriptor and see if that just works?

Reply:
done